kube-client/src/client/builder.rs

```diff
@@ -104,14 +107,36 @@ impl TryFrom<Config> for ClientBuilder<GenericService> {
             }
 
             #[cfg(feature = "http-proxy")]
-            Some(proxy_url) if proxy_url.scheme_str() == Some("http") => {
+            Some(proxy_url) if proxy_url.scheme_str() == Some(PROXY_SCHEME_HTTP) => {
```

Comment:
I just noticed that we only support HTTP proxies (rather than HTTPS), but that's probably out of scope for this PR.